src/FixedFrequencyAgent.cpp

```diff
@@ -0,0 +1,309 @@
+/*
+ * Copyright (c) 2015, 2016, 2017, 2018, 2019, Intel Corporation
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ *       notice, this list of conditions and the following disclaimer.
+ *
+ *     * Redistributions in binary form must reproduce the above copyright
+ *       notice, this list of conditions and the following disclaimer in
+ *       the documentation and/or other materials provided with the
+ *       distribution.
+ *
+ *     * Neither the name of Intel Corporation nor the names of its
+ *       contributors may be used to endorse or promote products derived
+ *       from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY LOG OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "FixedFrequencyAgent.hpp"
+
+#include <iostream>
+
+#include <cmath>
+#include <cassert>
+#include <algorithm>
+
+#include "geopm/PluginFactory.hpp"
+#include "geopm/PlatformIO.hpp"
+#include "geopm/PlatformTopo.hpp"
+#include "geopm/Helper.hpp"
+#include "geopm/Agg.hpp"
+
+#include <string>
+
+namespace geopm
+{
+    FixedFrequencyAgent::FixedFrequencyAgent():
+        FixedFrequencyAgent(platform_io(), platform_topo())
+    {
+    }
+
+    FixedFrequencyAgent::FixedFrequencyAgent(PlatformIO &plat_io, const PlatformTopo &topo)
+        : m_platform_io(plat_io)
+        , m_platform_topo(platform_topo())
+        , m_last_wait{{0, 0}}
+        , M_WAIT_SEC(0.05) // 50mS wait
+        , m_do_write_batch(false)
+        , m_is_adjust_initialized(false)
+        , m_signal_available({{"FREQUENCY_ACCELERATOR", {}},
+                              {"NVML::UTILIZATION_ACCELERATOR", {}},
+                              {"NVML::UTILIZATION_MEMORY", {}},
```

Comment:
```suggestion
```
These signals aren't required for this agent.  The power and energy consumption aren't actually required either, but are likely to be of interest to GPU fixed frequency runs so we may wish to keep them in the trace list.

Reply:
Fixed